### PR TITLE
Forward+

### PR DESCRIPTION
**Pull request by Tleety**
**Directional Light** now added to the editor and implemented in the engine.

Removed _World_ from _Renderer_.

Some small bugfixes.

### CODE REVIEW

File: resources/Shaders/ForwardPlus.frag.glsl [hunk 1]
```diff
@@ -104,27 +115,31 @@ void main()
...
-	{
+	for(int i = start; i < start + amount; i++) {
 		int l = int(LightIndex[i]);
-
-		LightResult result = CalcPointLight(V * PointLights.List[l].Position, PointLights.List[l].Radius, PointLights.List[l].Color, PointLights.List[l].Intensity, viewVec, position, normal, PointLights.List[i].Falloff);
-
+		LightSource light = LightSources.List[l];
+		LightResult result;
+
+		if(light.Type == 1) { // point
```
Comment (JockeBTH) [inline resources/Shaders/ForwardPlus.frag.glsl:124]: "if(light.Type == 1)" Enum?

Comment (Tleety) [inline resources/Shaders/ForwardPlus.frag.glsl:124]: Det är en Enum i cpp filen, men den blir tills int när man skickar in den i shadern.

File: resources/Shaders/ForwardPlus.frag.glsl [hunk 2]
```diff
@@ -104,27 +115,31 @@ void main()
...
+
+	//fragmentColor += Input.DiffuseColor;
 	fragmentColor += Input.DiffuseColor * (totalLighting.Diffuse + totalLighting.Specular) * texel * Color;
 	//fragmentColor += Input.DiffuseColor * (totalLighting.Diffuse) * texel * Color;
-	//fragmentColor += vec4(0.0, LightGrids.Data[currentTile].Amount/3.0, 0, 1);
+	//fragmentColor += Input.DiffuseColor + vec4(0.0, LightGrids.Data[currentTile].Amount/3, 0, 1);
 	//fragmentColor = texel * Input.DiffuseColor * Color;
-	if(int(gl_FragCoord.x)%16 == 0 || int(gl_FragCoord.y)%16 == 0 )
-	{
+	//fragmentColor += vec4(currentTile/3600.f, 0, 0, 1);
```
Comment (JockeBTH) [inline resources/Shaders/ForwardPlus.frag.glsl:139]: Remove? redundant?

**Comment by Stiffly** [comment]
Looks good! Clear to proceed!
mieksmall

File: resources/Shaders/ForwardPlus.frag.glsl (hunk 1 above)
Comment (Stiffly) [inline resources/Shaders/ForwardPlus.frag.glsl:124]: But the readabilty d00d? 
bt

**Pull request merged**